compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.cpp

```diff
@@ -620,6 +620,45 @@ getIGEMMGenericConvDetails(linalg::LinalgOp linalgOp) {
   auto filterMapGEMM =
       AffineMap::get(numParallelDims + numKDims, 0, filterDims, ctx);
 
+  // Compute the output permutation for the im2col tensor to match the
+  // dimension order of the input tensor.
```

Comment:
I still can't wrap my head around this: Why a layout consistent with input(global) will be better for performance? 

I understand that we need to pay the price of a bad layout, previously, we always write as [B, M, K] layout which is more friendly for MFMAs because there's no need to do any re-layout after ds_read, ds_write on the contrary is worse because a transpose need to happen before writting to LDS.

With the new approach we seem to be just doing the opposite, making ds_write easier but ds_read later worse. What is the essence of the improvement here? Do we reduce the bank conflict here?